### PR TITLE
NOTIF-820 Do not process endpoints that are not READY

### PR DESCRIPTION
**Pull request by gwenneg**
~Based on #1545.~ (rebased)

⚠️ This PR **MUST NOT** be merged before #1545 has been deployed on prod. ⚠️

### CODE REVIEW

File: engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java [hunk 1]
```diff
@@ -76,9 +78,11 @@ public Endpoint getOrCreateDefaultEmailSubscription(String accountId, String org
 
     public List<Endpoint> getTargetEndpoints(String orgId, EventType eventType) {
         String query = "SELECT DISTINCT e FROM Endpoint e JOIN e.behaviorGroupActions bga JOIN bga.behaviorGroup.behaviors b " +
-                "WHERE e.enabled IS TRUE AND b.eventType = :eventType AND (bga.behaviorGroup.orgId = :orgId OR bga.behaviorGroup.orgId IS NULL)";
+                "WHERE e.enabled IS TRUE AND e.status = :status AND b.eventType = :eventType " +
```
Comment (gwenneg) [inline engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java:81]: From now on, only `READY` endpoints will be used while processing an incoming ingress Kafka message.

**Comment by gwenneg** [comment]
Rebased onto `master` after the merge of #1545.

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

**Comment by gwenneg** [comment]
#1545 has been deployed on prod ✅

**Pull request merged**